src/main/java/gov/nih/nci/bento_ri/service/CPIFetcherService.java

```diff
@@ -0,0 +1,393 @@
+package gov.nih.nci.bento_ri.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import gov.nih.nci.bento_ri.model.*;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import com.github.benmanes.caffeine.cache.Cache;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Service for making OAuth2 authenticated API calls to CPI (Child Participant Index) service
+ */
+@Service("CpiFetcherService")
+public class CPIFetcherService {
+    
+    private static final Logger logger = LogManager.getLogger(CPIFetcherService.class);
+    
+    @Value("${cpi.oauth2.client.id:#{environment.OAUTH2_CLIENT_ID}}")
+    private String clientId;
+    
+    @Value("${cpi.oauth2.client.secret:#{environment.OAUTH2_CLIENT_SECRET}}")
+    private String clientSecret;
+    
+    @Value("${cpi.oauth2.token.uri:#{environment.OAUTH2_TOKEN_URI}}")
+    private String tokenUri;
+    
+    @Value("${cpi.api.url:#{environment.API_URL}}")
+    private String apiUrl;
+    
+    @Value("${cpi.domains.url:#{environment.DOMAINS_URL}}")
+    private String domainsUrl;
+    
+    @Value("${cpi.oauth2.scope:#{environment.OAUTH2_SCOPE}}")
+    private String scope;
```

Comment:
Hub hardcodes the fallbacks (the part after the `#`) for these three variables. I changed it for C3DC so that they try to read the env vars `API_URL`, `DOMAINS_URL`, and `OAUTH2_SCOPE`.